### PR TITLE
[TM-4684] Feature Flags Update

### PR DESCRIPTION
**Pull request by jlam-mpc**
Update:
- Added the following feature flags
  - Publishable Positions 
  - Projected Vacancy
  - Cycle Management
  - Bid Season Management (not attached to any components/routes yet)
  - Maintain Assignment (not attached to any components/routes yet)
  - Panel Admin (Run/Post Panel, Create/Edit Remarks, Create/Edit Panel Meeting)

[TM-4684](https://metaphase.atlassian.net/browse/TM-4684)

[TM-4684]: https://metaphase.atlassian.net/browse/TM-4684?atlOrigin=eyJpIjoiNWRkNTljNzYxNjVmNDY3MDlhMDU5Y2ZhYzA5YTRkZjUiLCJwIjoiZ2l0aHViLWNvbS1KU1cifQ

### CODE REVIEW

File: public/config/config.json [hunk 1]
```diff
@@ -7,13 +7,18 @@
...
     "deto": false,
     "handshake_bidding": true,
+    "maintain_assignment": true,
+    "panel_admin": true,
     "panel_meetings_search": true,
     "panel_meeting_agendas": true,
     "persona_auth": true,
     "post_handshake": false,
     "publishable_positions": true,
+    "projected_vacancy": true,
```
Comment (pveland621) [inline public/config/config.json:21]: :heart_eyes: at maintaining alphabetical order!

**Review by pveland621 — COMMENTED** [review]
Looks great! Pretty straightforward, there's only one thing I can think of that could possibly warrant any changes before merging:

- When we feature flag entire screens, I think hiding the menu option is enough, but just checking (@elizabeth-jimenez @scott062) that we don't want to add flag checks on the entire components themselves (as theoretically the user could enter in the URL and see unfinished pages, but that's such a fringe edge case, probably don't need to do that)

**Comment by jlam-mpc** [comment]
> Looks great! Pretty straightforward, there's only one thing I can think of that could possibly warrant any changes before merging:
> 
> * When we feature flag entire screens, I think hiding the menu option is enough, but just checking (@elizabeth-jimenez @scott062) that we don't want to add flag checks on the entire components themselves (as theoretically the user could enter in the URL and see unfinished pages, but that's such a fringe edge case, probably don't need to do that)

Oh this is a good point I thought about this too but got a little confused because some of these feature flags don't disable the whole component either so I thought we only needed to remove access from the UI? If I get confirmation I'll work on that~

File: src/sass/_administrator.scss [hunk 2]
```diff
@@ -459,4 +459,11 @@
       margin-top: 14px;
     }
   }
+
+  &.feature-flags-editor {
+    .jsoneditor-frame {
```
Comment (elizabeth-jimenez) [inline src/sass/_administrator.scss:464]: nice! I never even noticed that search bar 😂

File: src/Constants/Menu.js [hunk 3]
```diff
@@ -149,14 +149,15 @@ export const GET_PROFILE_MENU = () => MenuConfig([
...
       },
-      {
-        text: 'Panel',
-        route: '/profile/administrator/panel/',
-        icon: 'calendar',
-        roles: [
-          'superuser',
-        ],
-      },
+      checkFlag('flags.publishable_positions') ?
```
Comment (elizabeth-jimenez) [inline src/Constants/Menu.js:152]: I wonder if we want to put this behind its own flag 🤔

Comment (elizabeth-jimenez) [reply src/Constants/Menu.js:152]: oh panel_admin!

**Comment by elizabeth-jimenez** [comment]
> > Looks great! Pretty straightforward, there's only one thing I can think of that could possibly warrant any changes before merging:
> > 
> > * When we feature flag entire screens, I think hiding the menu option is enough, but just checking (@elizabeth-jimenez @scott062) that we don't want to add flag checks on the entire components themselves (as theoretically the user could enter in the URL and see unfinished pages, but that's such a fringe edge case, probably don't need to do that)
> 
> Oh this is a good point I thought about this too but got a little confused because some of these feature flags don't disable the whole component either so I thought we only needed to remove access from the UI? If I get confirmation I'll work on that~

Yea, the menu item is enough to hide the page, but we do need to add the feature flag check to wherever we link to these pages - for example the Add Panel Meeting button link on Panel Meeting Search page

File: src/Constants/Menu.js (hunk 3 above)
Comment (jlam-mpc) [reply src/Constants/Menu.js]: Done!

File: src/Constants/Menu.js [hunk 4]
```diff
@@ -187,24 +188,26 @@ export const GET_PROFILE_MENU = () => MenuConfig([
...
-      {
-        text: 'Cycle Management',
-        route: '/profile/bureau/cyclemanagement',
-        icon: 'cogs',
-        roles: [
-          'super_user',
-          'bureau_user',
-        ],
-      },
+      checkFlag('flags.publishable_positions') ?
```
[the remark this answers is not in the document]
Comment (jlam-mpc) [reply src/Constants/Menu.js]: Done!

**Comment by elizabeth-jimenez** [comment]
Agenda Item Maintenance Toggle to Edit button still needed

**Review by elizabeth-jimenez — APPROVED** [review]

**Pull request merged**